### ISSUE TITLE
TensorRT INT8 assertion failed when performing calibration

### ISSUE THREAD

**Issue by qinyao-he** [tensorflow/tensorflow#21998]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**:
source
- **TensorFlow version (use command below)**:
('v1.9.0-rc2-3472-g61038f8', '1.10.0')
github master commit: 61038f89e98fc430c56a9aa056692413c82eaafa

- **Python version**:
- **Bazel version (if compiling from source)**:
0.16.1
- **GCC/Compiler version (if compiling from source)**:
5.4.0
- **CUDA/cuDNN version**:
CUDA 9.0 cuDNN 7.1.4
- **GPU model and memory**:
Titan V
- **Exact command to reproduce**:
I build TensorFlow from source with TensorRT 4.

Use official TensorFlow-TensorRT example:
https://github.com/tensorflow/models/tree/master/research/tensorrt
and do INT8 test with command:

> python tensorrt.py --frozen_graph=resnetv2_imagenet_frozen_graph.pb --image_file=image.jpg --int8 --output_dir=output

Note that you should download ImageNet pretrained model http://download.tensorflow.org/models/official/resnetv2_imagenet_frozen_graph.pb

### Describe the problem
Program core dumped when doing calibration step

> INFO:tensorflow:Starting Warmup cycle
> 2018-08-31 14:53:15.227087: I tensorflow/contrib/tensorrt/kernels/trt_engine_op.cc:577] Starting calibration thread on device 0, Calibration Resource @ 0x7f76f0001860                             
> 2018-08-31 14:53:31.166463: I tensorflow/contrib/tensorrt/kernels/trt_engine_op.cc:577] Starting calibration thread on device 0, Calibration Resource @ 0x7f76f1709b00                             
> python: helpers.cpp:56: nvinfer1::DimsCHW nvinfer1::getCHW(const nvinfer1::Dims&): Assertion `d.nbDims >= 3' failed.                                                                               
> [1]    4880 abort (core dumped)  python tensorrt.py --frozen_graph=resnetv2_imagenet_frozen_graph.pb  --int8

This problem doesn't exist with tf 1.7 and 1.8.

**Comment by qinyao-he** [comment]
@aaroey

**Comment by tensorflowbutler** [comment]
Nagging Assignees @shivaniag, @aaroey: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tilaba** [comment]
have you solved this iusse? @qinyao-he

**Comment by aaroey** [comment]
Sorry for late reply. The assertion error is caused by a known bug in TRT 4.0 library, which should be fixed by TRT 5.0.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @shivaniag, @aaroey: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @shivaniag, @aaroey: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @shivaniag, @aaroey: It has been 61 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mohantym** [comment]
Hi @qinyao-he! 
It seems you are using older versions(1.x versions) of Tensorflow which is not supported anymore. Have you checked the above issue in latest version TF 2.6 yet?. Thanks!

**Comment by qinyao-he** [comment]
Should have been solved.

**Comment by mohantym** [comment]
Are you satisfied with the resolution of your issue?
[Yes](https://goo.gl/forms/Oe0tEvODFRoI2gJF3)
[No](https://goo.gl/forms/fUjzOfrtkFbrOT8d2)
